src/main/java/minijava/ir/optimize/UnreachableCodeRemover.java

```diff
@@ -7,11 +7,15 @@
 
   @Override
   public boolean optimize(Graph graph) {
-    // first remove Bad nodes we inserted manually (the remove_unreachable_code function doesn't pick them up)
+    //Dump.dumpGraph(graph, "before-removal");
+
     binding_irgopt.remove_bads(graph.ptr);
-    binding_irgopt.remove_unreachable_code(
-        graph.ptr); // find and replace unreachable code with Bad nodes
+    // find and replace unreachable code with Bad nodes
+    binding_irgopt.remove_unreachable_code(graph.ptr);
+
+    // checking whether a change on the graph occurred doesn't seem to be possible
     binding_irgopt.remove_bads(graph.ptr);
-    return false; // checking whether a change on the graph occurred doesn't seem to be possible
+
+    return false;
   }
 }
```

Comment:
probably :D I had played around with it because some stuff broke, but later on notice that wasn't broken any longer... so yeah, I'll clean that up